index.js

```diff
@@ -258,6 +275,12 @@ class TagInput<T> extends React.PureComponent<Props<T>, State> {
     scrollView.scrollTo({ y, animated: true });
   }
 
+  scrollToRight = () => {
+    setTimeout(() => {
+      this.scrollView.scrollToEnd({ animated: true });
```

Comment:
1. It looks like you still aren't running Flow to check for type errors. Please run Flow and correct the errors!
2. What's the difference between this and `scrollToBottom`? Can `scrollToBottom` just be reimplemented by calling `scrollToEnd`? If so, do we even need to keep track of both `this.contentHeight` and `this.scrollViewHeight` anymore?

Reply:
Brilliant! I merged `scrollToBottom` and `scrollToRight` it actually works and kinda fixed trivial deviation of previous vertical scroll. 
`this.scrollViewHeight` indeed could be taken out. However it seems `this.contentHeight` is essential for vertical scroll trigger `onScrollViewContentSizeChange`